version: '3.8'
services:
  postgres:
    image: postgres
    platform: linux/amd64
    volumes:
      - ./data/db:/var/lib/postgresql/data
    environment:
      - POSTGRES_DB=${POSTGRES_DB}
      - POSTGRES_USER=${POSTGRES_USER}
      - POSTGRES_PASSWORD=${POSTGRES_PASSWORD}
    container_name: reebelo_db
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U postgres"]
      interval: 5s
      timeout: 5s
      retries: 5
    ports:
      - '15432:5432'
    networks:
      - backend_network
  app:
    build: .
    platform: linux/amd64
    volumes:
      - .:/django
    ports:
      - '8000:8000'
    image: app:django
    container_name: reebelo_django
    command: python manage.py runserver 0.0.0.0:8000
    depends_on:
      postgres:
        condition: service_healthy
    networks:
      - backend_network
networks:
  backend_network:
    name: backend_network